libs/langchain/langchain/vectorstores/pinecone.py

```diff
@@ -52,16 +62,16 @@ def __init__(
                 "Could not import pinecone python package. "
                 "Please install it with `pip install pinecone-client`."
             )
-        if not isinstance(index, pinecone.index.Index):
-            raise ValueError(
-                f"client should be an instance of pinecone.index.Index, "
-                f"got {type(index)}"
-            )
         if not isinstance(embedding, Embeddings):
             warnings.warn(
                 "Passing in `embedding` as a Callable is deprecated. Please pass in an"
                 " Embeddings object instead."
             )
+        if not isinstance(index, pinecone.index.Index):
+            raise ValueError(
+                f"client should be an instance of pinecone.index.Index, "
```

Comment:
For my own understanding, is this to ensure it's an instance of our REST client vs our gRPC client?

Reply:
I think so. It's marked here in PR as added but this piece of code for checking index was here way before this PR.